lib/unpack.js

```diff
@@ -178,6 +178,13 @@ class Unpack extends Parser {
       if (parts.length < this.strip)
         return false
       entry.path = parts.slice(this.strip).join('/')
+
+      if (entry.type === 'Link') {
+        const linkparts = entry.linkpath.split(/\/|\\/)
+        /* istanbul ignore else */
+        if (!(linkparts.length < this.strip))
```

Comment:
There is a dedicated comparator to mean "not less-than".  Please use `if (linkparts.length >= this.strip)`.